test/e2e/network/service.go

```diff
@@ -924,37 +924,53 @@ var _ = SIGDescribe("Services", func() {
 			framework.ExpectNoError(err, "failed to delete service: %s in namespace: %s", serviceName, ns)
 		}()
 		jig.SanityCheckService(tcpService, v1.ServiceTypeClusterIP)
-		svcPort := int(tcpService.Spec.Ports[0].Port)
-		e2elog.Logf("service port TCP: %d", svcPort)
+		e2elog.Logf("Service Port TCP: %v", tcpService.Spec.Ports[0].Port)
 
-		// Change the services to NodePort and add a UDP port.
+		ginkgo.By("changing the TCP service to type=NodePort")
+		nodePortService := jig.UpdateServiceOrFail(ns, tcpService.Name, func(s *v1.Service) {
+			s.Spec.Type = v1.ServiceTypeNodePort
+			s.Spec.Ports = []v1.ServicePort{
+				{
+					Name:       "tcp-port",
+					Port:       80,
+					Protocol:   v1.ProtocolTCP,
+					TargetPort: intstr.FromInt(9376),
+				},
+			}
+		})
+
+		jig.CreateTCPUDPServicePods(cs, ns, 2)
+		execPod := e2epod.CreateExecPodOrFail(cs, ns, "execpod", nil)
+		jig.CheckServiceReachability(ns, nodePortService, execPod)
 
-		ginkgo.By("changing the TCP service to type=NodePort and add a UDP port")
-		newService := jig.UpdateServiceOrFail(ns, tcpService.Name, func(s *v1.Service) {
+		ginkgo.By("Updating NodePort service to listen TCP and UDP based requests over same Port")
+		nodePortService = jig.UpdateServiceOrFail(ns, tcpService.Name, func(s *v1.Service) {
 			s.Spec.Type = v1.ServiceTypeNodePort
 			s.Spec.Ports = []v1.ServicePort{
 				{
-					Name:     "tcp-port",
-					Port:     80,
-					Protocol: v1.ProtocolTCP,
+					Name:       "tcp-port",
+					Port:       80,
+					Protocol:   v1.ProtocolTCP,
+					TargetPort: intstr.FromInt(9376),
 				},
 				{
-					Name:     "udp-port",
-					Port:     80,
-					Protocol: v1.ProtocolUDP,
+					Name:       "udp-port",
+					Port:       80,
+					Protocol:   v1.ProtocolUDP,
+					TargetPort: intstr.FromInt(9376),
 				},
 			}
 		})
-		jig.SanityCheckService(newService, v1.ServiceTypeNodePort)
-		if len(newService.Spec.Ports) != 2 {
-			e2elog.Failf("new service should have two Ports")
+		jig.CheckServiceReachability(ns, nodePortService, execPod)
+
+		if nodePortCounts := len(nodePortService.Spec.Ports); nodePortCounts != 2 {
+			e2elog.Failf("new service should have two Ports but found %d Ports", nodePortCounts)
```

Comment:
nit: Here can be `framework.ExpectEqual(len(nodePortService.Spec.Ports), 2)`

Reply:
Done.